System/Random.hs

```diff
@@ -271,14 +271,18 @@ class RandomGen g where
   -- for more details. It is thus deprecated. If you need random
   -- values from other types you will need to construct a suitable
   -- conversion function.
+  --
+  -- Implementation note: the default implementation of 'next' ignores the
+  -- result of 'genRange'. It generates an 'Int' in the full domain of the
+  -- type. Instances of 'RandomGen' must implement both 'next' and 'genRange'
+  -- or neither. Implementing only 'genRange' but not 'next' or vice versa will
+  -- lead to unexpected results.
   next :: g -> (Int, g)
-  next g = (minR + fromIntegral w, g') where
-    (minR, maxR) = genRange g
-    range = fromIntegral $ maxR - minR
+  next g = (fromIntegral w, g') where
```

Comment:
This is only correct if `genRange = (minBound, maxBound)`

According to documentation of `next` it should return a value in the range of `genRange`

In other words this is totally wrong.

Also considering that we are deprecating `next` I don't see a need for its optimization

Reply:
> This is only correct if genRange = (minBound, maxBound)
>
> According to documentation of next it should return a value in the range of genRange
>
> In other words this is totally wrong.

Did you read the comment I added?

```
  -- Implementation note: the default implementation of 'next' ignores the
  -- result of 'genRange'. It generates an 'Int' in the full domain of the
  -- type. Instances of 'RandomGen' must implement both 'next' and 'genRange'
  -- or neither. Implementing only 'genRange' but not 'next' or vice versa will
  -- lead to unexpected results.
```

> Also considering that we are deprecating next I don't see a need for its optimization

That's a different point entirely. I am happy to revert if you feel this is unnecessary.

Reply:
I saw the comment, of course. And it directly contradicts the original documentation:

> The next operation returns an Int that is uniformly distributed in the range returned by genRange (including both end points), and a new generator.

Default implementation of a class function should be correct instead of fast. If it is the other way around then no warning/comment will save it ;)